design/ceph/object/store.md

```diff
@@ -176,3 +183,50 @@ spec:
   gateway:
     caBundleRef: #ldaps-cabundle
 ```
+
+### Virtual Host Style access for buckets
+
+The Ceph Object Gateway supports accessing buckets using [virtual host style](https://docs.aws.amazon.com/AmazonS3/latest/userguide/VirtualHosting.html) which allows accessing buckets using the bucket name as a subdomain in the endpoint. The user can configure this option manually like below:
+
+```sh
+# ceph config set client.rgw.mystore.a rgw_dns_name my-ingress.mydomain.com,rook-ceph-rgw-my-store.rook-ceph.svc
+```
+
+ The hostname should valid characters codified by RFC-1123, and Rook Operator can perform input validation using k8s apimachinery `IsDNS1123Subdomain()`. Multiple endpoints can be added to the list separated by comma. But when the support is added Rook operator need to pass this arguments to RGW daemon by the flag `--rgw-dns-name`. This is supported from Ceph Reef release(v18.0) onwards. This is different from `customEndpoints` which is used for configuring the object store to replicate and sync data amongst Multisite. For accessing the bucket point user need to configure wildcard dns in the cluster using [ingress loadbalancer](https://kubernetes.io/docs/concepts/services-networking/ingress/#hostname-wildcards) or in openshift cluster use [dns operator](https://docs.openshift.com/container-platform/4.9/networking/dns-operator.html). Same for TLS certificate, user need to configure the TLS certificate for the wildcard dns for the RGW endpoint. This option won't be enabled by default, user need to enable it by adding the `hosting` section in the `Gateway` settings:
```

Comment:
This is kind of a huge blob of ideas all on one line. Let's separate these for clarity. 

Also, we should capture the reason why we are using the `--rgw-dns-name` flag instead of using the mon config store.

```suggestion
Multiple hostnames can be added to the list separated by comma. Each entry must be a valid RFC-1123 hostname, and Rook Operator will perform input validation using k8s apimachinery `IsDNS1123Subdomain()`. 

When `rgw_dns_name` is changed for an RGW cluster, all RGWs need to be restarted. To enforce this in Rook, we can apply the `--rgw-dns-name` flag, which will restart RGWs with no user action needed. This is supported from Ceph Reef release(v18.0) onwards. 

This is different from `customEndpoints` which is used for configuring the object store to replicate and sync data amongst Multisite. 

For accessing the bucket point user need to configure wildcard dns in the cluster using [ingress loadbalancer](https://kubernetes.io/docs/concepts/services-networking/ingress/#hostname-wildcards) or in openshift cluster use [dns operator](https://docs.openshift.com/container-platform/4.9/networking/dns-operator.html). Same for TLS certificate, user need to configure the TLS certificate for the wildcard dns for the RGW endpoint. This option won't be enabled by default, user need to enable it by adding the `hosting` section in the `Gateway` settings:
```